tests/integration/nupic/algorithms/extensive_temporal_memory_test_cpp.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import unittest
+
+from nupic.bindings.algorithms import TemporalMemory
+
+from extensive_temporal_memory_test import ExtensiveTemporalMemoryTest
+
+
+
+if __name__ == "__main__":
+  ExtensiveTemporalMemoryTest.TM_CLASS = TemporalMemory
```

Comment:
This should be outside the `__main__` block (won't be executed by py.test otherwise). And we are moving away from the `__main__` block anyway so could omit it and `unittest.main`